#### Q1. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation of 5 using Python. Interpret the results.

In [2]:
import scipy.stats as stat

In [6]:
sample_mean=50
sample_std=5
sample_size=100


#standard error
std_error=sample_std/(sample_size**0.5)

#confidence interval
con_int=stat.norm.interval(0.95,loc=sample_mean,scale=std_error)
con_int

(49.02001800772997, 50.97998199227003)

#### Q2. Conduct a chi-square goodness of fit test to determine if the distribution of colors of M&Ms in a bag matches the expected distribution of 20% blue, 20% orange, 20% green, 10% yellow, 10% red, and 20% brown. Use Python to perform the test with a significance level of 0.05.

In [7]:
import numpy as np
import scipy.stats as  stat

In [10]:
##observer freq
obs_freq = np.array([20, 10, 20, 20, 10, 20])
ex_freq=np.array([20,20,20,10,10,20])

In [11]:
statistics,pvalue=stat.chisquare(obs_freq,ex_freq)

In [12]:
pvalue

0.010362337915786429

In [13]:
if pvalue < 0.05 :
    print("reject the null hypothesis")
else:
    print("Accept the null hypothesis")

reject the null hypothesis


#### Q3. Use Python to calculate the chi-square statistic and p-value for a contingency table with the following data:

In [ ]:
from scipy.stats import chisquare
import numpy as np

grp_a_exp = np.array([20, 10, 15])
grp_b_obs = np.array([15, 25, 20])

chi_sq, p_value = chisquare(f_obs=grp_b_obs, f_exp=grp_a_exp)

print("Chi-square statistic:", chi_sq)
print("P-value:", p_value)

#### Q4. A study of the prevalence of smoking in a population of 500 individuals found that 60 individuals smoked. Use Python to calculate the 95% confidence interval for the true proportion of individuals in the population who smoke.

In [3]:
import statsmodels.api as sm
import numpy as np

In [4]:
n=500
x=60
pro=x/n

In [6]:
## calculate the confidence interval
ci=sm.stats.proportion_confint(x,n,alpha=0.05)
ci

(0.09151638887130706, 0.14848361112869293)

#### Q5. Calculate the 90% confidence interval for a sample of data with a mean of 75 and a standard deviation of 12 using Python. Interpret the results.

In [12]:
import scipy.stats as stats

data = [50,75,100] 
mean = np.mean(data)
std_dev = np.std(data, ddof=1) 

n = len(data)
confidence_level = 0.90
alpha = 1 - confidence_level

t_critical = stats.t.ppf(1 - alpha/2, df=n-1)

margin_of_error = t_critical * (std_dev / np.sqrt(n))

lower_bound = mean - margin_of_error
upper_bound = mean + margin_of_error

print("90% confidence interval: ({:.2f}, {:.2f})".format(lower_bound, upper_bound))


90% confidence interval: (32.85, 117.15)


#### Q10. A study was conducted to determine if the proportion of people who prefer milk chocolate, dark chocolate, or white chocolate is different in the U.S. versus the U.K. A random sample of 500 people from the U.S. and a random sample of 500 people from the U.K. were surveyed. The results are shown in the contingency table below. Conduct a chi-square test for independence to determine if there is a significant association between chocolate preference and country of origin.

In [24]:
import scipy.stats as stat
observed=np.array([[200,150,150],[225,175,100]])
expected=np.array([[212.5,162.5,150],[212.5,162.5,150]])
df=(observed[0]-1)*(expected[0]-1)


In [36]:
chi2, pval, dof, exp_freq = stats.chi2_contingency(observed)
pval

0.0012348168997745918

In [ ]:
# p-value (0.123) is greater than the significance level of 0.05, we fail to reject the null hypothesis

### Q11. A random sample of 30 people was selected from a population with an unknown mean and standard deviation. The sample mean was found to be 72 and the sample standard deviation was found to be 10. Conduct a hypothesis test to determine if the population mean is significantly different from 70. Use a significance level of 0.05.

In [42]:
import math

sample_mean = 72
hypothesized_mean = 70
sample_std_dev = 10
sample_size = 30

t = (sample_mean - hypothesized_mean) / (sample_std_dev / math.sqrt(sample_size))
print("t-statistic:", t)


t-statistic: 1.0954451150103321


#### Q9. A study was conducted to determine if there is an association between smoking status (smoker or non-smoker) and lung cancer diagnosis (yes or no). The results are shown in the contingency table below. Conduct a chi-square test for independence to determine if there is a significant association between smoking status and lung cancer diagnosis.

In [53]:
observed=np.array([[60,140],[30,170]])

In [54]:
observed

array([[ 60, 140],
       [ 30, 170]])

In [55]:
from scipy.stats import chi2_contingency
stat, p, dof, expected = chi2_contingency(observed)

In [56]:
expected

array([[ 45., 155.],
       [ 45., 155.]])

In [59]:
import scipy.stats as stat
## chi square goodness of fit
chisquare_test,p_value=stat.chisquare(observed,expected)
p_value
chisquare_test

array([10.        ,  2.90322581])

In [58]:
siginifinace=0.05
dof=len(observed)-1
critivalvalue=stat.chi2.ppf(1-siginifinace,dof)
critivalvalue

3.841458820694124

we can reject the null hypothesis 
chisquare_test > critivalvalue

#### Q8. A researcher hypothesizes that a coin is biased towards tails. They flip the coin 100 times and observe  45 tails. Conduct a chi-square goodness of fit test to determine if the observed frequencies match the expected frequencies of a fair coin. Use a significance level of 0.05.

Step 1: Define the null and alternative hypotheses:

H0: The coin is fair (i.e., P(heads) = 0.5, P(tails) = 0.5)
H1: The coin is biased towards tails (i.e., P(heads) > 0.5, P(tails) < 0.5)

Step 2: Calculate the expected frequencies under the null hypothesis:

Under the null hypothesis, we expect the number of heads and tails to be equal. Therefore, the expected frequencies are:

Expected frequency of heads: 0.5 * 100 = 50
Expected frequency of tails: 0.5 * 100 = 50
Step 3: Calculate the test statistic:

The test statistic for a chi-square goodness of fit test is calculated as:

X2 = Σ ((Oi - Ei)^2 / Ei)

where Oi is the observed frequency and Ei is the expected frequency.

In our case, the observed frequencies are 55 for heads and 45 for tails, and the expected frequencies under the null hypothesis are 50 for each. Therefore, the test statistic is:

X2 = ((55 - 50)^2 / 50) + ((45 - 50)^2 / 50) = 1

Step 4: Determine the p-value and compare it to the significance level:

To determine the p-value, we need to calculate the probability of observing a test statistic as extreme as the one we calculated (or more extreme) under the null hypothesis. We can use the chi-square distribution with 1 degree of freedom (since we have 2 categories - heads and tails - and we estimated 1 parameter - the probability of tails).

In [62]:
import scipy.stats as stats

p_value = 1 - stats.chi2.cdf(1, 1)

print(p_value)

0.31731050786291415


#### The p-value is approximately 0.317, which is greater than the significance level of 0.05. Therefore, we fail to reject the null hypothesis.